flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/AbstractDeploymentObserver.java

```diff
@@ -262,8 +262,11 @@ private void onMissingDeployment(FlinkDeployment deployment) {
      * @param context Context for reconciliation.
      */
     private void checkIfAlreadyUpgraded(FlinkDeployment flinkDep, Context context) {
-        Optional<Deployment> depOpt = context.getSecondaryResource(Deployment.class);
         var status = flinkDep.getStatus();
+        if (status.getReconciliationStatus().getLastReconciledSpec() == null) {
```

Comment:
Would could probably introduce an isFirstDeployment() getter to the DepoymentStatus. I found this check in the code in multiple places.

Reply:
+1, and I think the condition should be improve Since the lastReconciliationStatus will be materialized before the `deploy` so there is the case it was not really finished. We should take this into account.